File: container-images/app-wizard/internal/pr/claim.go
package pr

import (
	"fmt"

	"sigs.k8s.io/yaml"
)

const (
	appAPIVersion = "cloud.ogenki.io/v1alpha1"
	appKind       = "App"
)

// BuildClaimYAML assembles the App claim manifest from a request's parts. The
// metadata name is the app name; namespace comes from the stack registry.
func BuildClaimYAML(name, namespace string, spec map[string]any) ([]byte, error) {
	if spec == nil {
		spec = map[string]any{}
	}
	claim := map[string]any{
		"apiVersion": appAPIVersion,
		"kind":       appKind,
		"metadata": map[string]any{
			"name":      name,
			"namespace": namespace,
		},
		"spec": spec,
	}
	b, err := yaml.Marshal(claim)
	if err != nil {
		return nil, fmt.Errorf("marshal claim: %w", err)
	}
	return b, nil
}

// BuildKustomizationYAML generates the app-directory kustomization referencing
// app.yaml.
func BuildKustomizationYAML() ([]byte, error) {
	k := map[string]any{
		"apiVersion": "kustomize.config.k8s.io/v1beta1",
		"kind":       "Kustomization",
		"resources":  []any{"app.yaml"},
	}
	b, err := yaml.Marshal(k)
	if err != nil {
		return nil, fmt.Errorf("marshal kustomization: %w", err)
	}
	return b, nil
}

File: container-images/app-wizard/ui/src/form/claim.ts
// Assemble the App claim object from form state and dump it to YAML (FR-012).
import yaml from "js-yaml";
import { prune } from "./model";

export interface ClaimInput {
  name: string;
  namespace?: string;
  spec: unknown;
}

export function buildClaim({ name, namespace, spec }: ClaimInput): Record<string, unknown> {
  const metadata: Record<string, unknown> = { name: name || "<app-name>" };
  if (namespace) metadata.namespace = namespace;
  return {
    apiVersion: "cloud.ogenki.io/v1alpha1",
    kind: "App",
    metadata,
    spec: prune(spec) ?? {},
  };
}

export function claimToYaml(input: ClaimInput): string {
  return yaml.dump(buildClaim(input), { noRefs: true, lineWidth: 100, sortKeys: false });
}
